samples/javascript_nodejs/49.qnamaker-all-features/dialogs/qnamakerBaseDialog.js

```diff
@@ -27,10 +27,11 @@ class QnAMakerBaseDialog extends QnAMakerDialog {
      * Core logic of QnA Maker dialog.
      * @param {QnAMaker} qnaService A QnAMaker service object.
      */
-    constructor(knowledgebaseId, authkey, host) {
-        var noAnswer = ActivityFactory.fromObject(DefaultNoAnswer);
+    constructor(knowledgebaseId, authkey, host, defaultAnswer) {
+
+        var defaultAnswerActivity = !!defaultAnswer ? ActivityFactory.fromObject(defaultAnswer) : ActivityFactory.fromObject(DefaultAnswer);
```

Comment:
We might need to check for the empty string as well?

Reply:
added checks. Thanks.